crates/sui-replay/src/replay.rs

```diff
@@ -1419,6 +1430,33 @@ impl LocalExec {
             .await
     }
 
+    fn add_config_objects_if_needed(
+        &self,
+        status: &SuiExecutionStatus,
+    ) -> Vec<(ObjectID, SequenceNumber)> {
+        match parse_effect_error_for_denied_coins(status) {
+            Some(coin_type) => {
+                let Some(mut config_id_and_version) = self.config_and_versions.clone() else {
+                    panic!("Need to specify the config object ID and version for '{coin_type}' in order to replay this transaction");
+                };
+                // NB: the version of the deny list object doesn't matter
```

Comment:
Does it matter more for other config objects? I thought the relevance of its actual version was the same in both cases.

Reply:
Config objects are read directly via `get_object` (so latest version) for execution checks, so we need to provide the version for them when replaying since they will not appear in V1 effects. The version of the config object does matter, as an address could be denied in one epoch, but later be removed from the deny list in a later epoch, and then transaction could be replay. If we just supply a random config object (or went for the most recent version) it may be in a different state to when the transaction ran and the replay will fail.

Reply:
The way I interpreted this comment was that the version doesn't matter for the deny list config object but it does for other objects, but I thought the manner in which the latest version's object affected replay was the same in both cases, and that's what I wanted to confirm.